modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -446,6 +477,75 @@ PinMeshGenerator::PinMeshGenerator(const InputParameters & parameters)
   generateMetadata();
 }
 
+void
+PinMeshGenerator::generateFlexibleAssemblyBoundaries()
+{
+  SubdomainName outermost_block_name;
+  bool has_single_mesh_interval;
+
+  // Assemblies that invoke this method are either homogenized or have a single pin. First check if
+  // the assembly only has a single region. Otherwise, determine the outermost region for deletion
+  if (_homogenized || (_intervals.size() == 1))
+  {
+    outermost_block_name = "RGMB_PIN" + std::to_string(_pin_type) + "_R0";
+    has_single_mesh_interval = true;
+  }
+  else
+  {
+    outermost_block_name =
+        "RGMB_PIN" + std::to_string(_pin_type) + "_R" + std::to_string(_intervals.size() - 1);
+    has_single_mesh_interval = false;
+
+    // Invoke BlockDeletionGenerator to delete outermost mesh interval of assembly
+    auto params = _app.getFactory().getValidParams("BlockDeletionGenerator");
+
+    params.set<std::vector<SubdomainName>>("block") = {outermost_block_name};
+    params.set<MeshGeneratorName>("input") = _homogenized ? name() + "_2D" : name() + "_delbds";
+
+    addMeshSubgenerator("BlockDeletionGenerator", name() + "_del_outer", params);
+  }
+
+  {
+    // Invoke FlexiblePatternGenerator to triangulate deleted mesh interval
+    auto params = _app.getFactory().getValidParams("FlexiblePatternGenerator");
+
+    if (has_single_mesh_interval)
+      params.set<std::vector<MeshGeneratorName>>("inputs") = {};
+    else
+    {
+      params.set<std::vector<MeshGeneratorName>>("inputs") = {name() + "_del_outer"};
+      params.set<std::vector<libMesh::Point>>("extra_positions") = {libMesh::Point(0, 0, 0)};
+      params.set<std::vector<unsigned int>>("extra_positions_mg_indices") = {0};
+    }
+    params.set<bool>("use_auto_area_func") = true;
```

Comment:
I haven't gotten around to playing around with the auto area parameters to decide options to expose here, but yes I'm hoping this can be added as part of a future PR